external/fv3fit/fv3fit/keras/_models/shared/pure_keras.py

```diff
@@ -164,3 +165,23 @@ def dump(self, path: str) -> None:
                         }
                     )
                 )
+
+    def _renamed_dict_compatible_model(self) -> tf.keras.Model:
+        """
+        Rename keras model inputs/outputs to match input_variables and
+        output_variables, and ensure model accepts and outputs dicts.
+        """
+        renamed_inputs = [
+            # skip the first placeholder dim when specifying shape
```

Comment:
Is this necessary?

Reply:
Yes, the Input layer doesn't know to ignore the None dim so it adds a second placeholder dimension if you use the full shape of the original input.

Reply:
I actually wrote a func for this: https://github.com/ai2cm/fv3net/blob/0d539792b7ee9804e47b0cdb684e1e03682fe5d8/external/fv3fit/fv3fit/keras/adapters.py#L71